### ISSUE TITLE
Support pre-release versions in constraints specified via the "requires" section

### ISSUE THREAD

**Issue by bforchhammer** [loft-sh/devspace#1826]
**What happened?**  

We have a "minimum devspace version" specified via the `requires` section in our devspace configs, which works well for normal semantic versions but unfortunately doesn't allow us to use pre-release versions of devspace (which would be useful eg. to test bugfixes, like https://github.com/loft-sh/devspace/issues/1818 😉 ):

```
❯ devspace --version
devspace version 5.18.1-beta.0
❯ devspace deploy
[info]   Using namespace 'test'
[info]   Using kube context 'docker-desktop'
[fatal]  require versions: DevSpace version mismatch: 5.18.1-beta.0 (currently installed) does not match >= 5.17.0 (required by config). Please make sure you have installed DevSpace with version >= 5.17.0
```

**What did you expect to happen instead?**  

I expect `5.18.1-beta.0` to match the constraint of `>= 5.17.0`. 

**How can we reproduce the bug?** (as minimally and precisely as possible)  

My devspace.yaml:
```
version: v1beta11

require:
  devspace: ">= 5.17.0"
...
```

**Local Environment:**  
- DevSpace Version: 5.18.1-beta.0
- Operating System: mac
- ARCH of the OS: AMD64

/kind bug

**Comment by pratikjagrut** [comment]
@bforchhammer thanks for creating this issue. We'll add support for pre-release versions.

**Comment by bforchhammer** [comment]
Awesome, thanks for the quick fix! I can confirm it works with `v5.18.1-beta.1`! 🥳
